src/Compilers/Core/AnalyzerDriver/AnalyzerManager.cs

```diff
@@ -265,12 +265,6 @@ internal bool IsSupportedDiagnostic(DiagnosticAnalyzer analyzer, Diagnostic diag
 
             foreach (var diag in supportedDiagnostics)
             {
-                if (HasNotConfigurableTag(diag.CustomTags))
```

Comment:
@mavasani @srivatsn @tmeschter I think this is not right. now user can configure diagnostic that are marked as NotConfigurable.

I think we should do

 if (HasNotConfigurableTag(diag.CustomTags))
{
                    if (diag.IsEnabledByDefault) return false else continue;
}

we can't let diagnostic that has NotConfigurable reach diagnosticOptions.Contains ...